origins_workflow/src/Controller/ModerationStateController.php

```diff
@@ -144,6 +144,11 @@ private function transitionAllowed($entity, String $new_state) {
         $transition_allowed = TRUE;
       }
     }
+    elseif (($current_state == 'draft') && ($new_state == 'archived')) {
+      if ($current_user->hasPermission('use editorial transition archive')) {
+        $transition_allowed = TRUE;
+      }
+    }
```

Comment:
This permission check was missing as it wasn't previously possible to change state from 'draft' to 'archived'